doc/source/redis-memory-management.rst

```diff
@@ -0,0 +1,103 @@
+Redis Memory Management (EXPERIMENTAL)
+==============================================
+
+Ray stores metadata associated to tasks and objects in one or more Redis
+servers, as described in :doc:`internals-overview`.  Applications that are
+long-running or have high task/object generation rate could risk high memory
+pressure, potentially leading to out-of-memory (OOM) errors.
+
+Here, we describe an experimental feature that transparently flushes metadata
+entries out of Redis memory.
+
+Requirements
+-------------
+
+As of early July 2018, the automatic memory management feature requires building
+Ray from source.  We are planning on eliminating this step in the near future by
+releasing official wheels.
+
+Building Ray
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+First, follow `instructions to build Ray from source
+<installation.html#building-ray-from-source>`__ to install prerequisites.  After
+the prerequisites are installed, instead of doing the regular ``pip install`` as
+referenced in that document,
+
+.. code-block:: bash
+
+  git clone https://github.com/ray-project/ray.git
+  cd ray/python
+  pip install -e . --verbose  # Add --user if you see a permission denied error.
```

Comment:
Why include this line? You don't actually want people to run this line, right? You want people to run the line below.

Reply:
Done, good call